magento/pwa-studio
# Set cart page is loading while fetching data from network

## Description

This PR sets the cart page's `isCartUpdating` flag while we are fetching data from the network.

This will not allow users to "Proceed to Checkout" until we know we have the latest data from the network and the user is seeing the most up-to-date information.

Notably, this PR does *not* solve the cart page flash problem - the cart page _purposefully_ shows data returned from its cache first, and updates it with data fetched from the network. It purposefully does not show a loading spinner while we wait for network data (that would defeat the purpose of having cached data).

The "flash" is the page showing the cached data first and then switching to the network data.

PWA-321 will properly add items to the cache and actually solve the flash problem, but even then there is a chance the cache and the network are out of sync, and the changes in this PR are useful to have in place regardless.

## Related Issue





Closes PWA-416.

## Acceptance 



### Verification Stakeholders

### Specification



### Verification Steps


1. Add an item to your cart
1. Change your browser's network settings to "Fast 3G"
1. Navigate to the `/cart` page
1. See that the "Proceed to Checkout" button is disabled until the "get cart details" network call returns

## Screenshots / Screen Captures (if appropriate)

## Checklist

* I have updated the documentation accordingly, if necessary.
